### PR TITLE
Add unit tests for MEASUREMENT_EXTENSION_LOG

### PR DESCRIPTION
**Pull request by Xiaohanjlll**

### CODE REVIEW

**Comment by Zhiqiang520** [comment]
it looks good to me.

**Comment by jyao1** [comment]
`libspdm_responder_measurement_extension_log_test_main()` should be covered by `LIBSPDM_ENABLE_CAPABILITY_MEL_CAP`, not `LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP`

**Comment by Xiaohanjlll** [comment]
> `libspdm_responder_measurement_extension_log_test_main()` should be covered by `LIBSPDM_ENABLE_CAPABILITY_MEL_CAP`, not `LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP`

Thanks, I have changed to `LIBSPDM_ENABLE_CAPABILITY_MEL_CAP`.

File: unit_test/test_spdm_requester/get_measurement_extension_log.c [hunk 1]
```diff
@@ -431,6 +492,179 @@ libspdm_return_t libspdm_requester_get_measurement_extension_log_test_receive_me
...
+        transport_header_size = LIBSPDM_TEST_TRANSPORT_HEADER_SIZE;
+        spdm_response = (void *)((uint8_t *)*response + transport_header_size);
+
+        spdm_response->header.spdm_version = SPDM_MESSAGE_VERSION_13;
+        spdm_response->header.request_response_code = SPDM_MEASUREMENT_EXTENSION_LOG;
+        spdm_response->header.param1 = 0;
+        spdm_response->header.param2 = 0;
+        spdm_response->portion_length = LIBSPDM_MAX_MEL_BLOCK_LEN;
+        /* remainder_length is greater than the SPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE */
+        spdm_response->remainder_length = SPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE + 1;
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/get_measurement_extension_log.c:651]: `SPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE + 1` is `0`.

File: unit_test/test_spdm_requester/get_measurement_extension_log.c [hunk 2]
```diff
@@ -14,6 +14,63 @@ uint8_t m_libspdm_mel_test[LIBSPDM_MAX_MEASUREMENT_EXTENSION_LOG_SIZE];
 size_t m_libspdm_mel_len;
 uint8_t m_libspdm_mel_number;
 
+void generate_mel_entry_test()
+{
+    spdm_measurement_extension_log_dmtf_t *measurement_extension_log;
+    uint32_t mel_index;
+    spdm_mel_entry_dmtf_t *mel_entry;
+    size_t mel_entry_zise;
```
Comment (steven-bellock) [inline unit_test/test_spdm_requester/get_measurement_extension_log.c:22]: `_zise` -> `_size`

File: unit_test/test_spdm_responder/measurement_extension_log.c [hunk 3]
```diff
@@ -79,6 +103,258 @@ void libspdm_test_responder_measurement_extension_log_case1(void **state)
     assert_memory_equal(spdm_response + 1, spdm_mel, spdm_mel_len);
 }
 
+/**
+ * Test 2: request.lenth is less than the MEL len
```
Comment (steven-bellock) [inline unit_test/test_spdm_responder/measurement_extension_log.c:107]: `lenth` -> `length`

File: unit_test/test_spdm_responder/measurement_extension_log.c [hunk 4]
```diff
@@ -79,6 +103,258 @@ void libspdm_test_responder_measurement_extension_log_case1(void **state)
...
+                     SPDM_MEASUREMENT_EXTENSION_LOG);
+    assert_int_equal(spdm_response->header.param1, 0);
+    assert_int_equal(spdm_response->portion_length, spdm_mel_len);
+    assert_int_equal(spdm_response->remainder_length, 0);
+    assert_memory_equal(spdm_response + 1, (void *)spdm_mel,
+                        response_size - sizeof(spdm_measurement_extension_log_response_t));
+}
+
+/**
+ * Test 4: request.offset > spdm mel len , wrong request messag
```
Comment (steven-bellock) [inline unit_test/test_spdm_responder/measurement_extension_log.c:234]: `messag` -> `message`

Comment (Xiaohanjlll) [reply unit_test/test_spdm_responder/measurement_extension_log.c]: Thanks, I have already modified these problems.

File: unit_test/test_spdm_responder/measurement_extension_log.c (hunk 3 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_responder/measurement_extension_log.c]: Thanks, I have already modified these problems.

File: unit_test/test_spdm_requester/get_measurement_extension_log.c (hunk 2 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_requester/get_measurement_extension_log.c]: Thanks, I have already modified these problems.

File: unit_test/test_spdm_requester/get_measurement_extension_log.c (hunk 1 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_requester/get_measurement_extension_log.c]: Thanks, I have already modified these problems.

**Pull request merged**